db/memtable.cc

```diff
@@ -374,14 +374,12 @@ InternalIterator* MemTable::NewIterator(const ReadOptions& read_options,
 }
 
 InternalIterator* MemTable::NewRangeTombstoneIterator(
-    const ReadOptions& read_options, Arena* arena) {
-  assert(arena != nullptr);
+    const ReadOptions& read_options) {
   if (read_options.ignore_range_deletions) {
-    return NewEmptyInternalIterator(arena);
+    return NewEmptyInternalIterator();
```

Comment:
yeah i have an upcoming diff with a bunch of microoptimizations, need a few more minutes :)